### PR TITLE
feat: add @auth/astro framework library with example & docs

### PR DESCRIPTION
**Pull request by theotterlord**
> _NOTE_:
>
> - It's a good idea to open an issue first to discuss potential changes.
> - Please make sure that you are _NOT_ opening a PR to fix a potential security vulnerability. Instead, please follow the [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md) to disclose the issue to us confidentially.

## ☕️ Reasoning

:rocket: Introduces an Astro framework package, with documentation & an example. The package exposes Astro components in addition to the methods.

Thanks to @tajkirkpatrick and @nowaythatworked who've helped out with building, debugging & documenting this package.

## 🧢 Checklist

- [x] Documentation
- [ ] Tests
- [x] Ready to be merged

## 🎫 Affected issues

Please [scout and link issues](https://github.com/nextauthjs/next-auth/issues) that might be solved by this PR.

Fixes: INSERT_ISSUE_LINK_HERE

## 📌 Resources

- [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md)
- [Contributing guidelines](https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md)
- [Code of conduct](https://github.com/nextauthjs/.github/blob/main/CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

File: packages/frameworks-astro/index.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  // @ts-expect-error import.meta.env is used by Astro
+  authConfig.secret ??= import.meta.env.AUTH_SECRET
+  // @ts-expect-error
+  authConfig.trustHost ??= !!(
+    // @ts-expect-error
+    import.meta.env.env.AUTH_TRUST_HOST ??
+    // @ts-expect-error
+    import.meta.env.env.VERCEL ??
+    // @ts-expect-error
+    import.meta.env.env.NODE_ENV !== "production"
```
Comment (tajkirkpatrick) [inline packages/frameworks-astro/index.ts:83]: Probably meant to be only one set of `.env`s

**Comment by TheOtterlord** [comment]
We've made a few fixes & changes since I opened the pull request, and with the latest commit this new package now supports node `17.4.0` and higher. Overall the package is a lot more stable now, and is ready for review.

Can't wait to hear the feedback from the team & get this merged! :rocket:

**Comment by TheOtterlord** [comment]
I've deployed the included example application to https://auth-astro-example.vercel.app showing what it looks like.

**Comment by TheOtterlord** [comment]
Closed automatically when fixing merge conflicts -_-

See #6463 for the up-to-date PR

**Pull request closed without merging**